src/internal/operators/windowToggle.ts

```diff
@@ -56,152 +58,90 @@ import { lift } from '../util/lift';
-    super._complete();
-  }
-
-  unsubscribe() {
-    if (!this.closed) {
-      const { contexts } = this;
-      this.contexts = null!;
-      if (contexts) {
-        const len = contexts.length;
-        let index = -1;
-        while (++index < len) {
-          const context = contexts[index];
-          context.window.unsubscribe();
-          context.subscription.unsubscribe();
+  return (source: Observable<T>) =>
+    lift(source, function (this: Subscriber<Observable<T>>, source: Observable<T>) {
+      const subscriber = this;
+      const windows: Subject<T>[] = [];
+
+      const handleError = (err: any) => {
+        while (0 < windows.length) {
+          windows.shift()!.error(err);
         }
-      }
-      super.unsubscribe();
-    }
-  }
+        subscriber.error(err);
+      };
 
-  notifyNext(outerValue: any, innerValue: any): void {
-    if (outerValue === this.openings) {
-      let closingNotifier;
+      let openNotifier: Observable<O>;
       try {
-        const { closingSelector } = this;
-        closingNotifier = closingSelector(innerValue);
-      } catch (e) {
-        return this.error(e);
-      }
-
-      const window = new Subject<T>();
-      const subscription = new Subscription();
-      const context = { window, subscription };
-      this.contexts.push(context);
-      const innerSubscription = innerSubscribe(closingNotifier, new ComplexInnerSubscriber(this, context, 0));
-
-      if (innerSubscription!.closed) {
-        this.closeWindow(this.contexts.length - 1);
-      } else {
-        (<any>innerSubscription).context = context;
-        subscription.add(innerSubscription);
+        openNotifier = from(openings);
+      } catch (err) {
+        subscriber.error(err);
+        return;
       }
-
-      this.destination.next(window);
-    } else {
-      this.closeWindow(this.contexts.indexOf(outerValue));
-    }
-  }
-
-  notifyError(err: any): void {
-    this.error(err);
-  }
-
-  notifyComplete(inner: Subscription): void {
-    if (inner !== this.openSubscription) {
-      this.closeWindow(this.contexts.indexOf((<any>inner).context));
-    }
-  }
-
-  private closeWindow(index: number): void {
-    if (index === -1) {
-      return;
-    }
-
-    const { contexts } = this;
-    const context = contexts[index];
-    const { window, subscription } = context;
-    contexts.splice(index, 1);
-    window.complete();
-    subscription.unsubscribe();
-  }
+      openNotifier.subscribe(
+        new OperatorSubscriber(
+          subscriber,
+          (openValue) => {
+            const window = new Subject<T>();
+            windows.push(window);
+            const closingSubscription = new Subscription();
+            const closeWindow = () => {
+              arrRemove(windows, window);
+              window.complete();
+              closingSubscription.unsubscribe();
+            };
+            const closingSubscriber = new OperatorSubscriber(subscriber, closeWindow, handleError, closeWindow);
```

Comment:
nitpick: Creating the `new OperatorSubscriber` here - rather than inline - doesn't make this more readable, IMO, as the `subscribe` call is quite a bit further down. And the variable isn't used elsewhere.